File: app/src/main/java/com/example/getfood/ui/orderlist/OrderListDisplayAdapter.java
package com.example.getfood.ui.orderlist;

import android.content.Context;
import android.content.Intent;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.getfood.R;
import com.example.getfood.ui.orderdetail.OrderDetailActivity;
import com.fazemeright.canteen_app_models.models.FullOrder;

import java.util.List;

public class OrderListDisplayAdapter extends ListAdapter<FullOrder, OrderListDisplayAdapter.ViewHolder> {

    private Context context;

    OrderListDisplayAdapter(Context context) {
        super(new OrderListDiffCallBack());
        this.context = context;
    }

    void swapData(List<FullOrder> newList) {
        submitList(newList);
    }

    @NonNull
    @Override
    public OrderListDisplayAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.order_list_display_view_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final OrderListDisplayAdapter.ViewHolder holder, int position) {
        FullOrder item = getItem(position);

        holder.bind(item);
    }

    static class OrderListDiffCallBack extends DiffUtil.ItemCallback<FullOrder> {

        @Override
        public boolean areItemsTheSame(@NonNull FullOrder order, @NonNull FullOrder t1) {
            return order.getOrderId().equals(t1.getOrderId());
        }

        @Override
        public boolean areContentsTheSame(@NonNull FullOrder order, @NonNull FullOrder t1) {
            return order.equals(t1);
        }
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView orderIDTextView, orderAmountTextView, orderTimeTextView;

        ViewHolder(View itemView) {
            super(itemView);
            orderIDTextView = itemView.findViewById(R.id.orderIDTextView);
            orderAmountTextView = itemView.findViewById(R.id.orderAmountTextView);
            orderTimeTextView = itemView.findViewById(R.id.orderTimeTextView);
        }

        void bind(final FullOrder item) {

            String orderId;
            if (item.getDisplayID() != null &&
                    !item.getDisplayID().isEmpty()) {
                orderId = item.getDisplayID();
            } else {
                orderId = item.getOrderId();
            }
            orderIDTextView.setText(String.format("Order ID: %s", orderId));
            orderAmountTextView.setText(String.format("Amount: ₹ %s",
                    item.getOrderAmount()));
            orderTimeTextView.setText(item.getTimeToDeliver());

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent i = new Intent(context, OrderDetailActivity.class);
                    i.putExtra("TestOrderData", item);
                    context.startActivity(i);
                }
            });
        }
    }
}

File: app/src/main/java/com/example/getfood/ui/orderlist/OrderListDisplayAdapter.kt
package com.example.getfood.ui.orderlist

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.example.canteen_app_models.models.FullOrder
import com.example.getfood.R
import com.example.getfood.ui.orderdetail.OrderDetailActivity

class OrderListDisplayAdapter internal constructor(private val context: Context) : ListAdapter<FullOrder, OrderListDisplayAdapter.ViewHolder>(OrderListDiffCallBack()) {
    fun swapData(newList: List<FullOrder?>?) {
        submitList(newList)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.order_list_display_view_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = getItem(position)
        holder.bind(item)
    }

    internal class OrderListDiffCallBack : DiffUtil.ItemCallback<FullOrder>() {
        override fun areItemsTheSame(order: FullOrder, t1: FullOrder): Boolean {
            return order.orderId == t1.orderId
        }

        override fun areContentsTheSame(order: FullOrder, t1: FullOrder): Boolean {
            return order == t1
        }
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private var orderIDTextView: TextView = itemView.findViewById(R.id.orderIDTextView)
        private var orderAmountTextView: TextView = itemView.findViewById(R.id.orderAmountTextView)
        private var orderTimeTextView: TextView = itemView.findViewById(R.id.orderTimeTextView)
        fun bind(item: FullOrder?) {
            val orderId: String? = if (item!!.displayID != null &&
                    item.displayID!!.isNotEmpty()) {
                item.displayID
            } else {
                item.orderId
            }
            orderIDTextView.text = String.format("Order ID: %s", orderId)
            orderAmountTextView.text = String.format("Amount: ₹ %s",
                    item.orderAmount)
            orderTimeTextView.text = item.timeToDeliver
            itemView.setOnClickListener {
                val i = Intent(context, OrderDetailActivity::class.java)
                i.putExtra("TestOrderData", item)
                context.startActivity(i)
            }
        }

    }
}